posthog-core/src/index.ts

```diff
@@ -0,0 +1,460 @@
+import { assert, currentISOTime, currentTimestamp, generateUUID, removeTrailingSlash, retriable } from './utils'
+export * as utils from './utils'
+import { LZString } from './lz-string'
+import { SimpleEventEmitter } from './eventemitter'
+
+export abstract class PostHogCore {
+  // options
+  private apiKey: string
+  private host: string
+  private flushAt: number
+  private flushInterval: number
+  private captureMode: 'form' | 'json'
+  private sendFeatureFlagEvent: boolean
+  private flagCallReported: { [key: string]: boolean } = {}
+
+  // internal
+  private _events: SimpleEventEmitter
+  private _queue: PostHogQueueItem[]
+  private _flushTimer?: any
+
+  private _decideResponsePromise?: Promise<PostHogDecideResponse>
+  private _decideResponse?: PostHogDecideResponse
+  private _decideTimer?: any
+  private _decidePollInterval = 10000
+
+  // Abstract methods to be overridden by implementations
+  abstract fetch(url: string, options: PostHogFetchOptions): Promise<PostHogFetchResponse>
+  abstract getLibraryId(): string
+  abstract getLibraryVersion(): string
+  abstract getCustomUserAgent(): string | void
+  abstract setImmediate(fn: () => void): void
+
+  // This is our abstracted storage. Each implementation should handle its own
+  abstract getPersistedProperty(key: PostHogPersistedProperty): string | undefined
+  abstract setPersistedProperty(key: PostHogPersistedProperty, value: string): void
+
+  public enabled = true
+
+  constructor(apiKey: string, options: PosthogCoreOptions = {}) {
+    assert(apiKey, "You must pass your PostHog project's api key.")
+
+    this._queue = []
+    this.apiKey = apiKey
+    this.host = removeTrailingSlash(options.host || 'https://app.posthog.com')
+    this.flushAt = options.flushAt ? Math.max(options.flushAt, 1) : 20
+    this.flushInterval = options.flushInterval ?? 10000
+    this.captureMode = options.captureMode || 'form'
+    this.sendFeatureFlagEvent = options.sendFeatureFlagEvent ?? true
+    this._events = new SimpleEventEmitter()
+
+    // NOTE: It is important we don't initiate anything in the constructor as some async IO may still be underway on the parent
+    if (options.preloadFeatureFlags !== false) {
+      this.setImmediate(() => {
+        void this.reloadFeatureFlagsAsync()
+      })
+    }
+  }
+
+  protected getCommonEventProperties(): any {
+    return {
+      $lib: this.getLibraryId(),
+      $lib_version: this.getLibraryVersion(),
+    }
+  }
+
+  // NOTE: Props are lazy loaded from localstorage hence the complex getter setter logic
+  private get props() {
+    if (!this._props) {
+      this._props = JSON.parse(this.getPersistedProperty(PostHogPersistedProperty.Props) || '{}')
+    }
+    return this._props || {}
+  }
+
+  private set props(val: { [key: string]: any }) {
+    this._props = val
+  }
+
+  private _props:
+    | {
+        [key: string]: any
+      }
+    | undefined
+
+  enable() {
+    this.enabled = true
+  }
+
+  disable() {
+    this.enabled = false
```

Comment:
Shouldn't this stop any timers? E.g. flush interval or decide interval?

Reply:
Yeah, this is not finished yet. The web side of disabling / enabling is so much more complicated so I'm leaving this till the rest of the lib is stable to see how much crossover there is between the platform 👍